sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -2555,7 +2556,7 @@ def GetDatabaseAccount(
             url_connection = self.url_connection
 
         headers = base.GetHeaders(self, self.default_headers, "get", "", "", "",
-                                  documents._OperationType.Read,{})
+                                  documents._OperationType.Read,{}, client_id=str(uuid.uuid4()))
```

Comment:
This is wrong - it would generate an new UUID for each call to GetDatabaseAccount. The intent is to generate a stable UUID for a CosmsoClient - then that stable ID should be send for all GetDatabaseAccount calls - this would allow Gateway to randomly decide for new clientIds whether to use global or regional endpoint as write location - but for known clientIds keep returning sticky values. Since client talks to multiple Gateway nodes the implementation will need to use a form of hashing - to make sure same clientId value reliably results in the same decision (global vs. regional)

Reply:
Blocking on this.

Reply:
for this bit we would want to create a new variable in the client_connection instance of the CosmosClient (which is one per client, and only needs to be set on init) - we can decide there if it is worthwhile to be cached or not within the client_connection, but from there we can directly pass it to the HeadersPolicy that gets initialized with the CosmosClient. All this policy does is add any headers it has to every request.